src/PostManager/PostManagerInterface.php

```diff
@@ -2,21 +2,25 @@
 
 namespace Drupal\social_post\PostManager;
 
-use Drupal\social_api\BaseManager\BaseManagerInterface;
+use Drupal\social_api\OAuth2Manager\OAuth2ManagerInterface;
 
 /**
  * Defines an PostManager Interface.
  *
  * @package Drupal\social_post\PostManager
  */
-interface PostManagerInterface extends BaseManagerInterface {
+interface PostManagerInterface extends OAuth2ManagerInterface {
 
   /**
-   * Execute the api to post on social network.
+   * Wrapper for post method.
    *
-   * @return mixed
-   *   The user data.
+   * @param string $access_token
+   *   The access token.
+   * @param string $access_token_secret
+   *   The access token secret.
+   * @param string $status
+   *   The tweet text.
    */
-  public function doExecute();
+  public function doPost($access_token, $access_token_secret, $status);
```

Comment:
I don't think this method is necessary. As I told you before, you can not assume the Social Post implementer would need this exact parameters to post. With OAuth 2 you may only need the access token, with OAuth 1 you may need both access token and access token secret, etc. You can not assume anything here. Social API should be appropriate for every authentication protocol